Sage/carbon
# feat: add support for manually overriding stacking order of modals FE-6259

fix #6402, fix #6410

### Proposed behaviour


Surfaces `topModalOverride` prop on `Dialog`, `DialogFullScreen`, `Confirm`, `Alert` and `Sidebar` components to support manually overriding the stacking of them in the `ModalManager`. Ensures that this is always set for `Toast` component so that they can be closed even when a new modal opens after it in the DOM.

### Current behaviour


Stacking order of open modals cannot be overridden.
Toasts cannot be interacted with when a modal opens after it.

### Checklist




#### QA


### Additional context



### Testing instructions



The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.


Basic mocked MFE sandbox, I've imported various modals from two different versions of carbon to test this out. QA can mess around with different compositions as they see fit

https://codesandbox.io/s/loving-mestorf-jzwqld
